transcom/mymove
# [ MB-16029 ] Allow editing SIT for future SIT

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16029)

## Summary

This patch fixes `sitStatus` values returned by the server so that TOOs can edit
both active and inactive SIT service items. With this change, SIT status is returned in the MTO Shipment payload even when SIT start date is in the future. There were some small tweaks that needed to get made to the ShipmentSITDisplay, SubmitSITExtensionModal, and ReviewSITExtensionModal to help calculate the SIT end date to take into account the start of the SIT when it's in the future.

## Verification Steps for Reviewers

Please use the `SITFUT` move code to verify this functionality, but also check `PENDNG` and `SITEXT` to ensure that there are no regressions in functionality.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

This code is deployed in our ephemeral environment and can be accessed via this Slack link 🔒 https://trussworks.slack.com/archives/C01T40WC4G1/p1683912581352379
